File: Array_Functions/Reversing_an_array(in_place)/reverse_in_place.java
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class Main {
    public static <T> void reverseInPlace(List<T> list) {
        int start = 0;
        int end = list.size() - 1;

        while (start < end) {
            Collections.swap(list, start, end);
            start++;
            end--;
        }
    }

    public static void main(String[] args) {
        List<Integer> list = Arrays.asList(1, 2, 3);
        reverseInPlace(list);
        for (int i : list) {
            System.out.print(i + " ");
        }
        System.out.println();
    }
}

File: Array_Functions/Reversing_an_array(in_place)/reverse_in_place.py
def reverse_in_place(vc):
    start = 0
    end = len(vc) - 1

    while start < end:
        vc[start], vc[end] = vc[end], vc[start]
        start += 1
        end -= 1

if __name__ == "__main__":
    vc = [1, 2, 3]
    reverse_in_place(vc)
    print(" ".join(map(str, vc)))

